tools/shape_tool/ShapeTool.cpp

```diff
@@ -0,0 +1,305 @@
+#include "../../../../lib/drawing/tile.cpp"
+#include "../../../../lib/tiles/common.cpp"
+
+#include "TileEmbeds.cpp"
+#include "TileWindow.cpp"
+#include "ShapeWindow.cpp"
+#include "TileShapes.cpp"
+
+const string EMBED_spr_icon_shape_tool = SPRITES_BASE + "icon_shape_tool.png";
+
+
```

Comment:
Formatting. I'm just going to drop this all here.

- Keep indentation on blank lines.
- Use symbols not words, e.g. `@X == null`, or `X && Y` instead of `X is null`, or `X and Y`.
- A single space around operators.
- Always use single quotes.
- Blank line at the start and end of classes inside the parentheses.
- Never more than one blank line.
- I don't use parentheses around `return`/`break`/`continue` if it's the only statement, but generally don't put it inline with the `if`.
- Prefer the `length` getter over the `size()` method for arrays.

I'm honestly pretty inconsistent so you can decide how fussy you want to be.